### ISSUE TITLE
Change asserts to raised exceptions.

### ISSUE THREAD

**Issue by caternuson** [adafruit/Adafruit_CircuitPython_TLC5947#10]
Replace asserts with appropriate raised exception with helpful message.

**Comment by ArthurDent62** [comment]
Can you point me to a library which I can use as coding reference?

**Comment by caternuson** [comment]
There are actually a few already in this library. Ex:
https://github.com/adafruit/Adafruit_CircuitPython_TLC5947/blob/master/adafruit_tlc5947.py#L190
Basically just replicate the conditional of the asserts and raise an exception if true.
```python
if some_check:
    raise SomeException("Some useful info about what went wrong.")
```
Note that asserts trigger when the condition is false, so you'll want to kind of "invert" the existing checks. Also, pick an appropriate exception type. Read through the available ones and pick one that applies:
https://docs.python.org/3/library/exceptions.html#built-in-exceptions
But don't worry about it too much. If we think the exception type should change, we'll let you know as part of code review for the PR, and the change would be trivial.

**Comment by ArthurDent62** [comment]
I think that `if / raise` is not pythonic - two other options come to my mind:

1. assert statement supports helpful message (quick to do, but also not so much pythonic):
```Java
assert 0 <= channel < _CHANNELS * self._n , \
        "'channel' out of range (0,%s)" % _CHANNELS * self._n -1 
```

2. using try/catch (pythonic but it is not so easy to spot the relationship between `byte_start` and `channel`:
```Java
    try:
        high_byte = self._shift_reg[byte_start]
        low_byte = self._shift_reg[byte_start+1]
    except IndexError:
        print("'channel' out of range(0,%s)" % _CHANNELS * self._n -1)
```
What do you think?

**Comment by caternuson** [comment]
We've been doing `if / raise` in other CircuitPython drivers. With 1. it's a similar approach, but doesn't allow using a different exception type. The problem with 2. is that you are handling the exception internally and not passing it on.

**Comment by ArthurDent62** [comment]
Thanks for your insight.
When done, I presume I shall provide the changes in a separate PR than the one currently ongoing, right?

**Comment by caternuson** [comment]
Yes please.

**Comment by caternuson** [comment]
Fixed with #15
